packages/nouns-contracts/contracts/libs/Noracle.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title A library used to maintain Nouns Auction House price history
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+library Noracle {
+    struct Observation {
+        // The block.timestamp when the auction was settled
+        uint256 blockTimestamp;
```

Comment:
let's do some packing here

`blockTimestamp` can be be saved in uint32 (year 2106) or uint34 if we want to be optimistic (year 2514)
`nounId` can be much smaller, uint16? uint32?
`amount` can probably be reduced too, uint80 can represent up to ~1.2M eth if I'm not mistaken

Reply:
ideally this could be just one storage slot, but I'm not sure if that works.
we need at least 160 bits for the address, so that leaves 96 bits, which I think is not enough for the rest:

smallest values would be:
timestamp ~ 32 bits
amount (if we don't reduce decimals): ~ 70 (< ~1k ETH)
nounId ~ 16 bits
initialized ~ 1 bit
so at least ~ 119 bits

Reply:
perhaps a good way to solve this challenge is to give up on plenty of decimals for ETH
instead of 18 decimals we could use say 8 decimals, then we can hold 1K ETH in 40 bits
wdyt?

Reply:
just pushed some bit packing changes and a test for the amount accuracy, lmk what you think

Reply:
I would prefer if we kept the original amount somehow.
let's keep this issue open for now

Reply:
yes, perhaps `blockTimestamp > 1` is the easiest condition
could also use nounId, but that's harder to reuse a new nouns fork

Reply:
changed here https://github.com/nounsDAO/nouns-monorepo/pull/598/commits/5468d13a2ceacac4cca2fd7266a5ecfb2fd0c52c